### PR TITLE
Import TWIR

### PR DESCRIPTION
**Pull request by vipulnsward**

### CODE REVIEW

File: _posts/2024-10-18-this-week-in-rails.markdown [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+published: true
+date: 2024-10-18
+---
+
+Hey everyone, Happy Friday! 
+
+[Vipul](https://www.saeloun.com/team/vipul) here with the latest updates for This Week in Rails. Let's dive in!
+
+[Rails World talks are out!](https://rubyonrails.org/2024/10/15/rails-world-2024-recap)  
+Check out recap of these talks in this [blog post](https://rubyonrails.org/2024/10/15/rails-world-2024-recap),
```
Comment (gregmolnar) [inline _posts/2024-10-18-this-week-in-rails.markdown:16]:
```suggestion
Check out the recap of these talks in this [blog post](https://rubyonrails.org/2024/10/15/rails-world-2024-recap), 
```

File: _posts/2024-10-18-this-week-in-rails.markdown [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+date: 2024-10-18
+---
+
+Hey everyone, Happy Friday! 
+
+[Vipul](https://www.saeloun.com/team/vipul) here with the latest updates for This Week in Rails. Let's dive in!
+
+[Rails World talks are out!](https://rubyonrails.org/2024/10/15/rails-world-2024-recap)  
+Check out recap of these talks in this [blog post](https://rubyonrails.org/2024/10/15/rails-world-2024-recap), 
+or head over [to Rails YouTube](https://www.youtube.com/watch?v=-cEn_83zRFw&list=PLHFP2OPUpCeb182aDN5cKZTuyjn3Tdbqx) for the full playlist.
```
Comment (gregmolnar) [inline _posts/2024-10-18-this-week-in-rails.markdown:17]:
```suggestion
or head over [to Rails' YouTube](https://www.youtube.com/watch?v=-cEn_83zRFw&list=PLHFP2OPUpCeb182aDN5cKZTuyjn3Tdbqx) for the full playlist.
```

File: _posts/2024-10-18-this-week-in-rails.markdown [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+[New Rails maintenance policy and end of maintenance announcements](https://rubyonrails.org/2024/10/15/new-maintenance-policy-and-eol-annouments)  
+These changes are designed to provide clarity on support timelines and help to plan Rails upgrades effectively. 
+Full details of the new policy can be found on the [Rails site.](https://rubyonrails.org/maintenance)
```
Comment (gregmolnar) [inline _posts/2024-10-18-this-week-in-rails.markdown:21]:
```suggestion
Full details of the new policy can be found on the [Rails website.](https://rubyonrails.org/maintenance)
```

File: _posts/2024-10-18-this-week-in-rails.markdown [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+[Rails Versions 6.1.7.9, 7.0.8.5, 7.1.4.1, and 7.2.1.1 have been released!](https://rubyonrails.org/2024/10/15/Rails-Versions-7-0-8-5-7-1-4-1-and-7-2-1-1-have-been-released)  
+These are security patches addressing 4 possible ReDoS (Regular expression Denial of Service) attacks.
+All of these only affect Ruby versions below 3.2 so if you are using any of these old Rubies, its time to update Rails versions at your earliest convenience.
```
Comment (gregmolnar) [inline _posts/2024-10-18-this-week-in-rails.markdown:25]:
```suggestion
All of these only affect Ruby versions below 3.2, so if you are using any of these old Rubies, its time to update Rails versions at your earliest convenience.
```

File: _posts/2024-10-18-this-week-in-rails.markdown [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+[Fix db:migrate:reset task](https://github.com/rails/rails/pull/53240)  
+Before this fix `db:migrate:reset` was broken, because we started to load the schema before migrating. This prevented migrations from running. 
+This patch empties the schema before migrating, ensuring that migrations run as expected.
+
+[Improve ActionController::TestCase to expose a binary encoded request.body](https://github.com/rails/rails/pull/53298)  
+This change improve `ActionController::TestCase` to expose a binary encoded `request.body`. The rack spec clearly states:
```
Comment (gregmolnar) [inline _posts/2024-10-18-this-week-in-rails.markdown:32]:
```suggestion
This change improves `ActionController::TestCase` to expose a binary encoded `request.body`. The rack spec clearly states:
```

File: _posts/2024-10-18-this-week-in-rails.markdown [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+> The input stream is an IO-like object which contains the raw HTTP POST data.
+> When applicable, its external encoding must be “ASCII-8BIT” and it must be opened in binary mode.
+
+Until now its encoding was generally UTF-8, which doesn't accurately reflect production behavior.
```
Comment (gregmolnar) [inline _posts/2024-10-18-this-week-in-rails.markdown:37]:
```suggestion
Until now, its encoding was generally UTF-8, which doesn't accurately reflect production behavior.
```

**Review by gregmolnar — COMMENTED** [review]
I found a few grammar improvements.

**Comment by vipulnsward** [comment]
Thanks! Applied

**Pull request merged**